engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/DefaultOAuth2ServiceFactory.scala

```diff
@@ -23,7 +29,24 @@ class DefaultOAuth2Service[ProfileResponse: Decoder](clientApi: OAuth2ClientApi[
     }
   }
 
-  def authorize(token: String): Future[LoggedUser] = clientApi.profileRequest(token).map(oAuth2Profile.getLoggedUser(_, configuration, allCategories))
+  def authorize(token: String): Future[LoggedUser] = getProfile(token).map(oAuth2Profile.getLoggedUser(_, configuration, allCategories))
+
+  private def getProfile(token: String): Future[ProfileResponse] = {
+    val profileFromJwtResult = Validated.fromOption(configuration.jwt.map(new JwtValidator[ProfileResponse](_)),
+      NonEmptyList.of(OAuth2JwtError("authentication.jwt configuration not provided"))) andThen { jwtValidator =>
+      jwtValidator.getProfileFromJwt(token)
+    }
+
+    /* Firstly checks whether a profile can be obtained from the token (provided authentication.jwt configured),
+     * secondly tries to obtain the profile from a sent request.
+     */
```

Comment:
I think if the user does not configure jwt, they should not get "authentication.jwt configuration not provided" error?